module/plugins/hooks/Captcha9Kw.py

```diff
@@ -1,254 +1,309 @@
-                time.sleep(5)
-            else:
-                break
+                if response2 == "" or response2 == "NO DATA":
+                    time.sleep(5)
+                else:
+                    break;
 
+            result = response2
+            task.data["ticket"] = response
+            self.log_info("result %s : %s" % (response, result))
+            task.setResult(result)
         else:
-            self.log_debug("Could not send request: %s" % res)
-            result = None
+            self.log_error("Bad upload: %s" % response)
+            return False
 
-        self.log_info(_("Captcha result for ticket %s: %s") % (res, result))
+    def newCaptchaTask(self, task):
+        if not task.isTextual() and not task.isPositional() and not task.isInteractive():
+            return False
 
-        task.setResult(result)
+        if not self.config.get("passkey"):
+            return False
 
-    def captcha_task(self, task):
-        if not task.isTextual() and not task.isPositional():
-            return
+        if self.pyload.isClientConnected() and not self.config.get("force"):
+            return False
 
-        if not self.config.get('passkey'):
-            return
+        if self.getCredits() > 0:
+            if self.config.get("queue") > 10 and self.config.get("queue") < 1000:
+                https = "https" if self.config.get("https") else "http"
+                servercheck = getURL(https + "://www.9kw.eu/grafik/servercheck.txt", get = {})
 
-        if self.pyload.isClientConnected() and self.config.get('check_client'):
-            return
+                for i in range(1, 3, 1):
+                    regex = re.compile("queue=(\d+)")
+                    r = regex.search(servercheck)
+                    if(self.config.get("queue") < r.group(1)):
+                        break;
 
-        credits = self.get_credits()
+                    time.sleep(10)
 
-        if not credits:
-            self.log_error(
-                _("Your captcha 9kw.eu account has not enough credits"))
-            return
-
-        queue = min(self.config.get('queue'), 999)
-        timeout = min(max(self.config.get('timeout'), 300), 3999)
-        pluginname = re.search(r'_(.+?)_\d+.\w+', task.captchaFile).group(1)
-
-        for _i in range(5):
-            servercheck = self.load("http://www.9kw.eu/grafik/servercheck.txt")
-            if queue < re.search(r'queue=(\d+)', servercheck).group(1):
-                break
-
-            time.sleep(10)
+            regex = re.compile("_([^_]*)_\d+.\w+")
+            r = regex.search(task.captchaFile)
+            pluginname = r.group(1)
 
+            if self.config.get("timeout") > 0:
+                timeout_option = self.config.get("timeout")
+            else:
+                timeout_option = 300
+
+            hosteroptions_conf = self.config.get("hosteroptions")
+            hosteroptions = hosteroptions_conf.split(";")
+
+            if len(hosteroptions) > 0:
```

Comment:
similar to before. this statement is superfluous due to the for loop afterwards.